### ISSUE TITLE
iptables moved in ubuntu-20 to /usr/sbin/iptables

### ISSUE THREAD

**Issue by georgejhunt** [iiab/iiab#2665]
But iiab-gen-iptables  in roles/network/templates/gateway/iiab-gen-iptable says:
{% if is_debuntu %}
IPTABLES=/sbin/iptables
IPTABLES_DATA=/etc/iptables.up.rules
{% else %}
IPTABLES=/usr/sbin/iptables
IPTABLES_DATA=/etc/sysconfig/iptables
{% endif %}

This surfaces because gateway_enabled is no longer working on my development machine. Maybe just a symbolic link from /sbin/iptables to /usr/sbin/iptables (that's what I did to fix in the short term)

**Comment by tim-moody** [comment]
on my x86 ub20.10 /sbin is linked to /usr/sbin and /sbin/iptables -S returns results

**Comment by holta** [comment]
@georgejhunt should this be marked as a bug?

**Comment by tim-moody** [comment]
I get the same result on my ub 20.04 vm as on my 20.10 nuc, i.e. that I don't see this problem.

**Comment by holta** [comment]
VM 191-u2004-srv-BIG-1226 = 10.8.0.50 is now installing -- feel free to log in there if it helps?

**Comment by holta** [comment]
> VM 191-u2004-srv-BIG-1226 = 10.8.0.50 is now installing -- feel free to log in there if it helps?

Just FYI on the above Ubuntu Server 20.04.1 (where BIG-sized IIAB has now finished installing) we have:

```
root@u20-srv:~# ls -l /sbin/iptables
lrwxrwxrwx 1 root root 26 Jul 31 12:29 /sbin/iptables -> /etc/alternatives/iptables
root@u20-srv:~# ls -l /usr/sbin/iptables
lrwxrwxrwx 1 root root 26 Jul 31 12:29 /usr/sbin/iptables -> /etc/alternatives/iptables
root@u20-srv:~# ls -l /bin/iptables
ls: cannot access '/bin/iptables': No such file or directory
root@u20-srv:~# ls -l /usr/bin/iptables
ls: cannot access '/usr/bin/iptables': No such file or directory

root@u20-srv:~# apt -a list iptables
Listing... Done
iptables/focal,now 1.8.4-3ubuntu2 amd64 [installed,automatic]

root@u20-srv:~# more /etc/issue.net
Ubuntu 20.04.1 LTS
root@u20-srv:~# uname -a
Linux box.lan 5.4.0-58-generic #64-Ubuntu SMP Wed Dec 9 08:16:25 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by holta** [comment]
@georgejhunt & @tim-moody: can you review proposed fix PR #2667, if you have time?

**Comment by tim-moody** [comment]
>This surfaces because gateway_enabled is no longer working on my development machine,

From @jvonau's analysis this is the primary issue and the missing symlink is not the root cause, but rather a change in the behavior of netfilter-persistent save causes the masquerade rule not to be saved. This is what #2667 addresses.

**Comment by holta** [comment]
Appears solved by PRs #2727 and #2728.

(Feel free to re-open as nec.)
